pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -319,3 +325,97 @@ func isLinuxGuest(guestID string) bool {
 
 	return strings.Contains(guestID, "Linux") || strings.Contains(guestID, "linux")
 }
+
+func logConfigSpec(
+	vmCtx context.VirtualMachineContextA2,
+	configSpec *vimTypes.VirtualMachineConfigSpec) {
+
+	if !pkgconfig.FromContext(vmCtx).LogFullBootstrapConfigAndCustomizationSpecs {
+		configSpec = SanitizeConfigSpec(configSpec)
+	}
+
+	vmCtx.Logger.Info("Customization Reconfigure", "configSpec", configSpec)
+}
+
+func SanitizeConfigSpec(configSpec *vimTypes.VirtualMachineConfigSpec) *vimTypes.VirtualMachineConfigSpec {
+	cs := *configSpec
```

Comment:
This is an exported function and there is an indirect without a nil check. Either make the param by-value or add the nil check.